neo4j-jdbc-bolt/src/main/java/org/neo4j/jdbc/bolt/BoltNeo4jPreparedStatement.java

```diff
@@ -79,7 +80,7 @@ public static PreparedStatement newInstance(boolean debug, BoltNeo4jConnectionIm
 
 	@Override public boolean execute() throws SQLException {
 		return executeInternal((result) -> {
-			boolean hasResultSet = hasResultSet();
+			boolean hasResultSet = hasResultSet((BoltNeo4jConnectionImpl) this.connection, this.statement);
```

Comment:
do you still need the cast since you added `newNeo4jSession` to `BoltNeo4jConnection`?

Reply:
I changed the cast to the interface, we still need it because `this.connection` is `Neo4jConnection`

Reply:
maybe only cast to `BoltNeo4jConnection`?